src/main/java/frc/robot/subsystems/TrapArmSubsystem.java

```diff
@@ -0,0 +1,358 @@
+    baseMotor1.getPIDController().setP(basePID[0]);
+    baseMotor1.getPIDController().setI(basePID[1]);
+    baseMotor1.getPIDController().setD(basePID[2]);
+    baseMotor1.getPIDController().setIZone(basePID[3]);
+    baseMotor1.getPIDController().setFF(basePID[4]);
+
+    baseMotor2 = new CANSparkMax(TrapArmConstants.baseMotor2_ID, MotorType.kBrushless);
+    baseMotor2.restoreFactoryDefaults();
+    baseMotor2.getPIDController().setP(basePID[0]);
+    baseMotor2.getPIDController().setI(basePID[1]);
+    baseMotor2.getPIDController().setD(basePID[2]);
+    baseMotor2.getPIDController().setIZone(basePID[3]);
+    baseMotor2.getPIDController().setFF(basePID[4]);
+
+    baseBreak = new DigitalInput(TrapArmConstants.baseBreak_ID);
+
+    scoringMotor = new CANSparkMaxSim(TrapArmConstants.scoringMotor_ID, MotorType.kBrushless);
+    scoringMotor.restoreFactoryDefaults();
+    scoringMotor.getPIDController().setP(scoringPID[0]);
+    scoringMotor.getPIDController().setI(scoringPID[1]);
+    scoringMotor.getPIDController().setD(scoringPID[2]);
+    scoringMotor.getPIDController().setIZone(scoringPID[3]);
+    scoringMotor.getPIDController().setFF(scoringPID[4]);
+
+    scoringBreak = new DigitalInput(TrapArmConstants.scoringBreak_ID);
+
+    // SmartDashboard
+    sourceLog = new DebugEntry<Boolean>(baseBreak.get(), "Source Beam Break", this);
+    groundLog = new DebugEntry<Boolean>(baseBreak.get(), "Ground Beam Break", this);
+    baseLog = new DebugEntry<Boolean>(baseBreak.get(), "Base Limit Switch", this);
+    scoringLog = new DebugEntry<Boolean>(baseBreak.get(), "Scoring Limit Switch", this);
+
+    basePIDEntry.setDoubleArray(basePID);
+    scoringPIDEntry.setDoubleArray(scoringPID);
+
+    // Simulation
+    if (Robot.isSimulation()) {
+      // 2D Mechanism
+      armMechanism = new Mechanism2d(4, 4);
+      root = armMechanism.getRoot("Root", 2, 0);
+      baseMech =
+          root.append(new MechanismLigament2d("Base Arm", 1.5, 90, 20, new Color8Bit(Color.kBlue)));
+      scoringMech =
+          baseMech.append(
+              new MechanismLigament2d("Scoring Arm", 4.5, 0, 10, new Color8Bit(Color.kAqua)));
+      wristMech =
+          scoringMech.append(
+              new MechanismLigament2d("Wrist Mech", 3, -170, 5, new Color8Bit(Color.kRed)));
+
+      m_baseElevatorSim =
+          new ElevatorSim(
+              DCMotor.getNEO(1),
+              70,
+              5.4,
+              Units.inchesToMeters(1),
+              0,
+              Units.inchesToMeters(18),
+              true,
+              0);
+
+      m_scoringElevatorSim =
+          new ElevatorSim(
+              DCMotor.getNEO(1),
+              70,
+              5.4,
+              Units.inchesToMeters(1),
+              0,
+              Units.inchesToMeters(18),
+              true,
+              0);
+
+      TrapArmTab.add("Trap Arm Mech", armMechanism);
+    } else {
+      m_baseElevatorSim = null;
+      m_scoringElevatorSim = null;
+    }
```

Comment:
I'd prefer removing this else clause and then remove the 'final' keyword from the instantiations so it doesn't complain when they stay unassigned.